### PR TITLE
Add docker support. Tested on RPI4

### PR DESCRIPTION
**Pull request by krisstakos**
## Description

Added docker support. 
Dockerfile for building the container and docker-compose for running it
Tested on Raspberry Pi 4

## Other information

Probably will run on many platforms without issues

### CODE REVIEW

**Comment by builder555** [comment]
this is a cool idea, most likely won't work on a Mac though, at least not without a lot more work - there's no dbus out of the box.

File: setup-dev.sh
Comment (builder555) [inline setup-dev.sh:1]: this will break on non-docker systems that don't have pipenv installed already

Comment (krisstakos) [reply setup-dev.sh:1]: Not really, we have section in the README:
For the backend script, first install:

    [git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
    [python 3](https://www.python.org/downloads/) (tested with 3.10/3.11)
    [pipenv](https://pipenv.pypa.io/en/latest/installation/)
    [node+npm](https://nodejs.org/en/download/)
    
 So we can make it clear that these are the per-requisites in order to build and run. (aka before even running anything from the repo). Also I'm not sure  why is that alias for pipenv, for me it was problematic. Once pipenv was installed it should be straight accessible without any aliases

**Comment by krisstakos** [comment]
> this is a cool idea, most likely won't work on a Mac though, at least not without a lot more work - there's no dbus out of the box.

Yeah, didn't think for the Mac, there is no way to even test it as I'm not in the eco system

File: setup-dev.sh
Comment (builder555) [reply setup-dev.sh:1]: if everyone read and followed the instructions, the entire documentation would've been 3 sentences long :)

**Pull request merged**